### DISCUSSION TITLE
"/cmi ride" bug with World Guard.

### DISCUSSION THREAD

**Discussion by BboyZeroOne** [Zrips/CMI#3144] [Enhancement]
**Description of issue:**
Players who use /cmi ride on non-vanilla ride-able mobs would such as chicken would be able to have access to restriction regions set with world guard. World Guard Flag: Entry-deny

---

**Cmi Version (using`/cmi version`):** 
8.6.22.2

**Server Type (Spigot/Paperspigot/etc):** 
PaperSpigot

**Server Version (using `/ver`):** 
Paper version git-Paper-183 

**Relevant plugins (Delete if this isn't needed):**
World Guard

**Personal Comment:**
I've talked to world guard dev and asked if anything could be done, this is his response.

"worldguard literally can't cancel events that don't happen
bukkit throws movement events for normal "player riding a vehicle" stuff
if we get playermove/vehiclemove events, we can cancel them"

WorldGuard Discord: https://discord.gg/sXRbMPb

**Comment by Zrips** [comment]
In general this is not a bug as this is issue with minecraft itself not firing player move event in case he is riding some entity and allowing player to be moved into areas where he should not be allowed to be in. Some complex work around needs to be added to resolve this issue, but there is no easy fix for it currently

**Comment by BboyZeroOne** [comment]
Alright, no worries. I'll wait for it to get fixed soon! No rush :)
